### PR TITLE
Re-enable testing on OSX 10.15

### PR DESCRIPTION
**Pull request by directhex**
Ref https://github.com/dotnet/runtime/pull/40523#issuecomment-670528185

### CODE REVIEW

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 1]
```diff
@@ -78,9 +78,7 @@ jobs:
       - ${{ if eq(parameters.jobParameters.isFullMatrix, true) }}:
         - OSX.1013.Amd64.Open
         - OSX.1014.Amd64.Open
-        # The 10.15 machines aren't in the same configuration, see
-        # https://github.com/dotnet/runtime/issues/24736
-        #- OSX.1015.Amd64.Open
+        - OSX.1015.Amd64.Open
```
Comment (safern) [inline eng/pipelines/libraries/helix-queues-setup.yml:81]: Should we do 1014 and 1015 on PRs and then the 3 on CI?

I feel like it is more important to have latest 2 on PRs. `isFullMatrix == false` is PRs.

**Comment by directhex** [comment]
I had to ActiveIssue https://github.com/dotnet/runtime/issues/40937 which is not Mac-specific, but only macOS 10.15 have a default OS image which exposes the bug

**Review by safern — APPROVED** [review]
Thanks, @directhex

**Pull request merged**